File: src/main/java/zd/zero/waifu/motivator/plugin/alert/WaifuMotivationFactory.java
package zd.zero.waifu.motivator.plugin.alert;

import com.intellij.openapi.project.Project;
import zd.zero.waifu.motivator.plugin.alert.notification.AlertConfiguration;

public interface WaifuMotivationFactory {

    WaifuMotivation constructMotivation( Project project, MotivationAsset motivation, AlertConfiguration config);

}

File: src/main/java/zd/zero/waifu/motivator/plugin/alert/WaifuMotivationFactory.kt
package zd.zero.waifu.motivator.plugin.alert

import com.intellij.openapi.project.Project
import zd.zero.waifu.motivator.plugin.alert.notification.AlertConfiguration

interface WaifuMotivationFactory {
    fun constructMotivation(project: Project, motivation: MotivationAsset, config: AlertConfiguration): WaifuMotivation
}
